packages/medusa/src/loaders/index.ts

```diff
@@ -196,5 +224,24 @@ export default async ({
     Logger.success(searchActivity, "Indexing event emitted") || {}
   track("SEARCH_ENGINE_INDEXING_COMPLETED", { duration: searchAct.duration })
 
-  return { container, dbConnection, app: expressApp, pgConnection }
+  /**
+   * One shouldn't have to know what resources need to be disposed
+   * under the hood. This function should be called when the server
+   * is shutting down.
+   */
+  async function disposeResources() {
```

Comment:
question: Will this be used for anything but tests? If not, I feel weird about having it in our main loader, but perhaps we have to.

Reply:
I thought that it could be useful for anybody starting manually for some reason, also, the main idea behind is that we remove the need for the user to have to know everything that needs to be disposed in any scenario. wdyt?

Reply:
this is only for tests. on the application shutdown, `process.exit` already cleans everything

Reply:
don't you think it could be useful for a user that want to start manually into something else? of if a user want to start it for its own tests?

Reply:
Should we remove it now and add it in case users request this feature? I haven't heard of anyone needing this up until this point

Reply:
you mean, I move that to our tests instead? cause we are using it but I can remove it from here and put it in the bootstrap, but we need to do that anyway for us

Reply:
I ll move it to the bootstrap in the test helpers instead.

Reply:
Actually, I'll let you two decide. My immediate thought was just that it was a bit misplaced